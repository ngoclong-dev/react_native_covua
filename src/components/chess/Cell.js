import React, { Component } from "react";
import PropTypes from "prop-types";
import { Text } from "react-native";
import styled from "styled-components/native";

const possibleMoveColor = "palegreen";
const selectedColor = "darkseagreen";
const regularLightColor = "ivory";
const regularDarkColor = "steelblue";

export const cellBackgroundColor = ({ col, row, possibleMove, selected }) => {
  const even = i => i % 2 === 0;
  const regularColor = even(col)
    ? even(row) ? regularLightColor : regularDarkColor
    : even(row) ? regularDarkColor : regularLightColor;
  return possibleMove
    ? possibleMoveColor
    : selected ? selectedColor : regularColor;
};
export const StyledCell = styled.TouchableHighlight`
  width: 40;
  height: 40;
  display: flex;
  flex-grow: 1;
  border-color: grey;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  background-color: ${cellBackgroundColor}
`;

const SymbolText = styled.Text`
  font-size: 24;
  color: black;
`;

export class Cell extends Component {
  static propTypes = {
    selected: PropTypes.bool,
    possibleMove: PropTypes.bool,
    onPress: PropTypes.func.isRequired,
    field: PropTypes.object.isRequired
  };

  onPress() {
    const { onPress, field } = this.props;
    onPress(field);
  }

  render() {
    const { field, selected, possibleMove } = this.props;
    const symbol = field.isEmpty ? "" : field.piece.toString();
    return (
      <StyledCell
        col={field.coordinates.col}
        row={field.coordinates.row}
        selected={selected}
        possibleMove={possibleMove}
        onPress={() => this.onPress()}
      >
        <SymbolText>{symbol}</SymbolText>
      </StyledCell>
    );
  }
}
