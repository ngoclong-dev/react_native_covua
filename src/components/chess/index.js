export { Game } from "./Game";
export { Board } from "./Board";
export { Cell } from "./Cell";