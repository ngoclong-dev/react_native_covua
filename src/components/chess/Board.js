import React, { Component } from "react";
import { Text, View } from "react-native";
import styled from 'styled-components/native';
import PropTypes from "prop-types";
import { Cell } from "./Cell";

const StyledBoard = styled.View`
  border: 1;
  width: 336;
  display: flex;
  flex-flow: row wrap;
  border-color: lightsteelblue;
  justify-content: space-around;
`;

export class Board extends Component {
  static propTypes = {
    board: PropTypes.object.isRequired,
    onCellPress: PropTypes.func.isRequired,
    selectedField: PropTypes.object,
    possibleMoves: PropTypes.array.isRequired,
  };

  render() {
    const { board, onCellPress, selectedField, possibleMoves } = this.props;
    const { fields } = board;
    const possibleFields = possibleMoves.map(({ to }) => to);

    return (
      <StyledBoard>
        {fields.map((field, index) => {
          const selected =
            !!selectedField &&
            selectedField.coordinates.index === field.coordinates.index;
          const possibleMove = possibleFields.some(
            f => f.index === field.coordinates.index
          );
          return (
            <Cell
              key={index}
              field={field}
              selected={selected}
              possibleMove={possibleMove}
              onPress={(field) => onCellPress(field)}
            />
          );
        })}
      </StyledBoard>
    );
  }
}
