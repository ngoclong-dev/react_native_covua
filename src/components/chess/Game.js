import React, { Component } from 'react';
import { Board } from './Board';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import * as actions from '../../flux/actions/gameActions';
import { restoreState } from '../../flux/middleware/saveState';

const enhance = connect(state => ({ game: state.game }));

const BoldText = styled.Text`
  font-weight: bold;
`;
const LightText = styled.Text`
  font-weight: 100;
`;

class Game extends Component {
  // async componentWillMount() {
  //   const { dispatch } = this.props;
  //   try {
  //     const { game } = await restoreState();
  //     if (game) {
  //       dispatch(actions.replaceBoard(game.board));
  //     }
  //   } catch (e) {}
  // }

  createGame() {
    const { dispatch } = this.props;
    dispatch(actions.newGame());
  }

  onCellPress(field) {
    const { dispatch } = this.props;
    dispatch(actions.selectFieldThunk(field));
  }

  render() {
    const { game } = this.props;
    const { board, selectedField, possibleMoves } = game;
    const thinking = game.thinking ? 'CPU' : 'Human';
    return (
      <View>
        <Board
          selectedField={selectedField}
          possibleMoves={possibleMoves}
          board={board}
          onCellPress={field => this.onCellPress(field)}
        />
        <BoldText>
          Move: <LightText>{thinking}</LightText>
        </BoldText>
      </View>
    );
  }
}

const ConnectedGame = enhance(Game);
export { ConnectedGame as Game };
