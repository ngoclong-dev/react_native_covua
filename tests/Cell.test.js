import React from "react";
import { Field, Coordinates } from "chess-js";
import { shallow } from "enzyme";
import { StyledCell, Cell, cellBackgroundColor } from "../src/components/chess/Cell";

const noop = () => {};

describe("<Cell />", () => {
  it("renders empty", () => {
    const coordinates = Coordinates.fromIndex(0);
    const field = new Field(coordinates, null);
    const wrapper = shallow(<Cell field={field} onPress={noop} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<StyledCell />", () => {
  it("renders unselected, non possible move", () => {
    const selected = false, possibleMove = false;
    const wrapper = shallow(
      <StyledCell
        selected={selected}
        possibleMove={possibleMove}
        onPress={noop}
      >
        X
      </StyledCell>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders selected, non possible move", () => {
    const selected = true, possibleMove = false;
    const wrapper = shallow(
      <StyledCell
        selected={selected}
        possibleMove={possibleMove}
        onPress={noop}
      >
        X
      </StyledCell>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders unselected, possible move", () => {
    const selected = false, possibleMove = true;
    const wrapper = shallow(
      <StyledCell
        selected={selected}
        possibleMove={possibleMove}
        onPress={noop}
      >
        X
      </StyledCell>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders selected, possible move", () => {
    const selected = true, possibleMove = true;
    const wrapper = shallow(
      <StyledCell
        selected={selected}
        possibleMove={possibleMove}
        onPress={noop}
      >
        X
      </StyledCell>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("cellBackgroundColor", () => {

  it("color when nothing is set", () => {
    const selected = false, possibleMove = false;
    const result = cellBackgroundColor({ selected, possibleMove });
    expect(result).toBe("ivory");
  });

  it("color when selected is set", () => {
    const selected = true, possibleMove = false;
    const result = cellBackgroundColor({ selected, possibleMove });
    expect(result).toBe("darkseagreen");
  });

  it("color when possibleMove is set", () => {
    const selected = false, possibleMove = true;
    const result = cellBackgroundColor({ selected, possibleMove });
    expect(result).toBe("palegreen");
  });

  it("color when both is set", () => {
    const selected = true, possibleMove = true;
    const result = cellBackgroundColor({ selected, possibleMove });
    expect(result).toBe("palegreen");
  });


});
