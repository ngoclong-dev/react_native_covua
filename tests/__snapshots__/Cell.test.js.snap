// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> renders empty 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(TouchableHighlight)
    col={0}
    onPress={[Function]}
    possibleMove={undefined}
    row={0}
    selected={undefined}
>
    <Styled(Text) />
</Styled(TouchableHighlight)>,
  "nodes": Array [
    <Styled(TouchableHighlight)
      col={0}
      onPress={[Function]}
      possibleMove={undefined}
      row={0}
      selected={undefined}
>
      <Styled(Text) />
</Styled(TouchableHighlight)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Cell
        field={
                Field {
                        "_coordinates": Coordinates {
                          "col": 0,
                          "row": 0,
                        },
                        "_piece": null,
                      }
        }
        onPress={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Cell {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "field": Field {
            "_coordinates": Coordinates {
              "col": 0,
              "row": 0,
            },
            "_piece": null,
          },
          "onPress": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(TouchableHighlight)
          col={0}
          onPress={[Function]}
          possibleMove={undefined}
          row={0}
          selected={undefined}
>
          <Styled(Text) />
</Styled(TouchableHighlight)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(TouchableHighlight)
          col={0}
          onPress={[Function]}
          possibleMove={undefined}
          row={0}
          selected={undefined}
>
          <Styled(Text) />
</Styled(TouchableHighlight)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Cell
    field={
        Field {
            "_coordinates": Coordinates {
              "col": 0,
              "row": 0,
            },
            "_piece": null,
          }
    }
    onPress={[Function]}
/>,
}
`;

exports[`<StyledCell /> renders selected, non possible move 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    possibleMove={false}
    selected={true}
    style={
        Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "darkseagreen",
              "borderColor": "grey",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            undefined,
          ]
    }
    underlayColor="black"
>
    X
</TouchableHighlight>,
  "nodes": Array [
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      possibleMove={false}
      selected={true}
      style={
            Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "darkseagreen",
                    "borderColor": "grey",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
      }
      underlayColor="black"
>
      X
</TouchableHighlight>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Styled(TouchableHighlight)
        onPress={[Function]}
        possibleMove={false}
        selected={true}
>
        X
</Styled(TouchableHighlight)>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledNativeComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "children": "X",
          "onPress": [Function],
          "possibleMove": false,
          "selected": true,
        },
        "refs": Object {},
        "state": Object {
          "generatedStyles": Object {
            "alignItems": "center",
            "backgroundColor": "darkseagreen",
            "borderColor": "grey",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={false}
          selected={true}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "darkseagreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
        "_debugID": 6,
        "_renderedOutput": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={false}
          selected={true}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "darkseagreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(TouchableHighlight)
    onPress={[Function]}
    possibleMove={false}
    selected={true}
>
    X
</Styled(TouchableHighlight)>,
}
`;

exports[`<StyledCell /> renders selected, possible move 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    possibleMove={true}
    selected={true}
    style={
        Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "palegreen",
              "borderColor": "grey",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            undefined,
          ]
    }
    underlayColor="black"
>
    X
</TouchableHighlight>,
  "nodes": Array [
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      possibleMove={true}
      selected={true}
      style={
            Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "palegreen",
                    "borderColor": "grey",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
      }
      underlayColor="black"
>
      X
</TouchableHighlight>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Styled(TouchableHighlight)
        onPress={[Function]}
        possibleMove={true}
        selected={true}
>
        X
</Styled(TouchableHighlight)>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledNativeComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "children": "X",
          "onPress": [Function],
          "possibleMove": true,
          "selected": true,
        },
        "refs": Object {},
        "state": Object {
          "generatedStyles": Object {
            "alignItems": "center",
            "backgroundColor": "palegreen",
            "borderColor": "grey",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={true}
          selected={true}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "palegreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
        "_debugID": 10,
        "_renderedOutput": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={true}
          selected={true}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "palegreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(TouchableHighlight)
    onPress={[Function]}
    possibleMove={true}
    selected={true}
>
    X
</Styled(TouchableHighlight)>,
}
`;

exports[`<StyledCell /> renders unselected, non possible move 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    possibleMove={false}
    selected={false}
    style={
        Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "ivory",
              "borderColor": "grey",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            undefined,
          ]
    }
    underlayColor="black"
>
    X
</TouchableHighlight>,
  "nodes": Array [
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      possibleMove={false}
      selected={false}
      style={
            Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "ivory",
                    "borderColor": "grey",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
      }
      underlayColor="black"
>
      X
</TouchableHighlight>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Styled(TouchableHighlight)
        onPress={[Function]}
        possibleMove={false}
        selected={false}
>
        X
</Styled(TouchableHighlight)>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledNativeComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "children": "X",
          "onPress": [Function],
          "possibleMove": false,
          "selected": false,
        },
        "refs": Object {},
        "state": Object {
          "generatedStyles": Object {
            "alignItems": "center",
            "backgroundColor": "ivory",
            "borderColor": "grey",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={false}
          selected={false}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "ivory",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
        "_debugID": 4,
        "_renderedOutput": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={false}
          selected={false}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "ivory",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(TouchableHighlight)
    onPress={[Function]}
    possibleMove={false}
    selected={false}
>
    X
</Styled(TouchableHighlight)>,
}
`;

exports[`<StyledCell /> renders unselected, possible move 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    possibleMove={true}
    selected={false}
    style={
        Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "palegreen",
              "borderColor": "grey",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
            undefined,
          ]
    }
    underlayColor="black"
>
    X
</TouchableHighlight>,
  "nodes": Array [
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      possibleMove={true}
      selected={false}
      style={
            Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "palegreen",
                    "borderColor": "grey",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
      }
      underlayColor="black"
>
      X
</TouchableHighlight>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Styled(TouchableHighlight)
        onPress={[Function]}
        possibleMove={true}
        selected={false}
>
        X
</Styled(TouchableHighlight)>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledNativeComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "children": "X",
          "onPress": [Function],
          "possibleMove": true,
          "selected": false,
        },
        "refs": Object {},
        "state": Object {
          "generatedStyles": Object {
            "alignItems": "center",
            "backgroundColor": "palegreen",
            "borderColor": "grey",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={true}
          selected={false}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "palegreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
        "_debugID": 8,
        "_renderedOutput": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          possibleMove={true}
          selected={false}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "palegreen",
                                "borderColor": "grey",
                                "display": "flex",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              },
                              undefined,
                            ]
          }
          underlayColor="black"
>
          X
</TouchableHighlight>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(TouchableHighlight)
    onPress={[Function]}
    possibleMove={true}
    selected={false}
>
    X
</Styled(TouchableHighlight)>,
}
`;
